lib/timers.js

```diff
@@ -232,6 +231,12 @@ function listOnTimeout() {
   } else if (list === refedLists[msecs]) {
     delete refedLists[msecs];
```

Comment:
I'm very confused as to why what I originally suggested doesn't do the job just fine?

```js
  if (list._unrefed === true && list === unrefedLists[msecs]) {
    delete unrefedLists[msecs];
    this.close();
  } else if (list === refedLists[msecs]) {
    delete refedLists[msecs];
    this.close();
  }
```

Keep in mind that `reuse()` is called before this if unrefing is actually re-using the handle and it already removed the list from the pool: https://github.com/nodejs/node/blob/75cdc895ec8fd38b3a22535d9f90792c7c7be964/lib/timers.js#L274-L293

Essentially I don't see why we'd bother to check `.owner()` if doing it in those conditionals is deterministic anyways?